src/internal/packager/git/gitea.go

```diff
@@ -120,6 +84,76 @@ func (g *Git) CreateReadOnlyUser() error {
 	return err
 }
 
+// UpdateReadOnlyUser uses the Gitea API to update a non-admin Zarf user.
+func (g *Git) UpdateReadOnlyUser(oldAdminPass string) error {
+	message.Debugf("git.UpdateReadOnlyUser()")
+
+	
+	c, err := cluster.NewCluster()
+	if err != nil {
+		return err
+	}
+	// Establish a git tunnel to send the repo
+	tunnel, err := c.NewTunnel(cluster.ZarfNamespaceName, k8s.SvcResource, cluster.ZarfGitServerName, "", 0, cluster.ZarfGitServerPort)
+	if err != nil {
+		return err
+	}
+	_, err = tunnel.Connect()
+	if err != nil {
+		return err
+	}
+	defer tunnel.Close()
+	tunnelURL := tunnel.HTTPEndpoint()
+ 
+	// Update the existing user's password
+	updateUserBody := map[string]interface{}{
+		"login_name": g.Server.PullUsername,
+		"password":   g.Server.PullPassword,
+	}
+	updateUserData, _ := json.Marshal(updateUserBody)
+	updateUserEndpoint := fmt.Sprintf("%s/api/v1/admin/users/%s", tunnelURL, g.Server.PullUsername)
+	updateUserRequest, _ := netHttp.NewRequest("PATCH", updateUserEndpoint, bytes.NewBuffer(updateUserData))
+	out, err := g.DoHTTPThings(updateUserRequest, g.Server.PushUsername, oldAdminPass)
+	message.Debugf("PATCH %s:\n%s", updateUserEndpoint, string(out))
+	return err
+}
+
+// UpdatePushUser uses the Gitea API to update an admin Zarf user.
+func (g *Git) UpdatePushUser(oldAdminPass string) error {
+	message.Debugf("git.UpdatePushUser()")
+
+	c, err := cluster.NewCluster()
+	if err != nil {
+		return err
+	}
+
+	// Establish a git tunnel to send the repo
+	tunnel, err := c.NewTunnel(cluster.ZarfNamespaceName, k8s.SvcResource, cluster.ZarfGitServerName, "", 0, cluster.ZarfGitServerPort)
+	if err != nil {
+		return err
+	}
+	_, err = tunnel.Connect()
+	if err != nil {
+		return err
+	}
+	defer tunnel.Close()
+
+	tunnelURL := tunnel.HTTPEndpoint()
+
+	
+	// Make sure the user can't create their own repos or orgs
+	updateUserBody := map[string]interface{}{
+		"login_name":      g.Server.PushUsername,
+		"password":        g.Server.PushPassword,
+	}
+	updateUserData, _ := json.Marshal(updateUserBody)
+	updateUserEndpoint := fmt.Sprintf("%s/api/v1/admin/users/%s", tunnelURL, g.Server.PushUsername)
+	updateUserRequest, _ := netHttp.NewRequest("PATCH", updateUserEndpoint, bytes.NewBuffer(updateUserData))
+	out, err := g.DoHTTPThings(updateUserRequest, g.Server.PushUsername, oldAdminPass)
+	message.Debugf("PATCH %s:\n%s", updateUserEndpoint, string(out))
+	return err
+}
+
```

Comment:
These functions should be combined into one that takes in the username to update

Reply:
refactored into a function that takes username and userpass for the update data. I guess we could do a conditional to check which username was passed in to decide the userpass to use if that's a more preferred convention.